### PR TITLE
Implement getCapabilities for iOS and finish skipToIndex

### PR DESCRIPTION
**Pull request by marwfair**
Issue: https://github.com/brim-borium/spotify_sdk/issues/102

This PR just implements `getCapabilities` for iOS and finishes the `skipToIndex` functionality for android and added it for iOS.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -104,6 +104,7 @@ Have a look [in the example](example/lib/main.dart) for detailed insights on how
 |  resume | Resumes the current track |✔ |  ✔ | ✔ |
 |  skipNext | Skips to next track | ✔ | ✔  | ✔ |
 |  skipPrevious | Skips to previous track |✔  |  ✔ | ✔ |
+|  skipToIndex | Skips to track at specified index in album or playlist |✔  |  ✔ | 🚧  |
```
Comment (fotiDim) [inline README.md:107]:
```suggestion
|  skipToIndex | Skips to track at specified index in album or playlist | 🚧 |  ✔ | 🚧 |
```
It is not done on Android right?

File: ios/Classes/PlayerState/PlayerState.swift [hunk 2]
```diff
@@ -68,4 +68,10 @@ struct State {
             "isEnabled": crossfadeState.isEnabled,
         ]
     }
+
+    static func userCapabilitiesDictionary(_ userCapabilities: SPTAppRemoteUserCapabilities) -> [String : Any] {
+        return [
+            "can_play_on_demand": userCapabilities.canPlayOnDemand
```
Comment (fotiDim) [inline ios/Classes/PlayerState/PlayerState.swift:74]:
```suggestion
            "canPlayOnDemand": userCapabilities.canPlayOnDemand
```

**Comment by fotiDim** [comment]
Thanks for the PR. I added some initial comments. I will have a deeper look by tomorrow.

File: README.md (hunk 1 above)
Comment (marwfair) [reply README.md:107]: It is done on android. The `skipToIndex` function was already there in SpotifyPlayerApi.kt. It was just missing the dart and swift implementations.

File: ios/Classes/PlayerState/PlayerState.swift (hunk 2 above)
Comment (marwfair) [reply ios/Classes/PlayerState/PlayerState.swift:74]: I'm pretty sure it has to be `can_play_on_demand`. `capabilities.dart` is using that as the json key. I also just gave it a try the other way and it didn't work.

**Comment by marwfair** [comment]
> Thanks for the PR. I added some initial comments. I will have a deeper look by tomorrow.

Awesome! Thank you!

File: ios/Classes/PlayerState/PlayerState.swift (hunk 2 above)
Comment (fotiDim) [reply ios/Classes/PlayerState/PlayerState.swift:74]: You are right. I just saw `isEnabled` above and I assumed all values were like that. Turns out we have an inconsistency there.

File: ios/Classes/SwiftSpotifySdkPlugin.swift [hunk 3]
```diff
@@ -204,6 +228,23 @@ public class SwiftSpotifySdkPlugin: NSObject, FlutterPlugin {
...
             }
             appRemote.userAPI?.removeItemFromLibrary(withURI: uri, callback: defaultPlayAPICallback)
+        case SpotfySdkConstants.methodGetCapabilities:
+            guard let appRemote = appRemote else {
+                result(FlutterError(code: "Connection Error", message: "AppRemote is null", details: nil))
+                return
+            }
+            appRemote.userAPI?.fetchCapabilities(callback: { (capabilitiesResult, error) in
+                guard error == nil else {
+                    result(FlutterError(code: "UserAPI Error", message: error?.localizedDescription, details: nil))
```
Comment (fotiDim) [inline ios/Classes/SwiftSpotifySdkPlugin.swift:238]:
```suggestion
                    result(FlutterError(code: "getCapabilitiesError", message: error?.localizedDescription, details: nil))
```
To be consistent with Android...

File: ios/Classes/SwiftSpotifySdkPlugin.swift [hunk 4]
```diff
@@ -204,6 +228,23 @@ public class SwiftSpotifySdkPlugin: NSObject, FlutterPlugin {
...
+                    return
+                }
+                guard let userCapabilities = capabilitiesResult as? SPTAppRemoteUserCapabilities else {
+                    result(FlutterError(code: "UserAPI Error", message: error?.localizedDescription, details: nil))
```
Comment (fotiDim) [inline ios/Classes/SwiftSpotifySdkPlugin.swift:242]:
```suggestion
                    result(FlutterError(code: "getCapabilitiesError", message: error?.localizedDescription, details: nil))
```
To be consistent with Android

**Review by fotiDim — APPROVED** [review]

**Pull request merged**